### PR TITLE
New Feature multilingual: Adding flags to articles info and list when associated

### PR DESCRIPTION
**Pull request by infograf768**
New feature:
Adding the possibility by parameter to display the associations next to the article.
First in Isis:
For blog, featured, single article and category list

screen shot 2016-08-31 at 12 38 51

screen shot 2016-08-31 at 12 47 27

### CODE REVIEW

**Comment by infograf768** [comment]
@andrepereiradasilva
Thanks for looking

File: components/com_content/views/article/tmpl/default.php [hunk 1]
```diff
@@ -18,7 +18,18 @@
...
+
+// Check if associations are implemented. If they are, define the parameter.
+if (JLanguageAssociations::isEnabled())
+{
+	$assocParam = $params->get('show_associations');
+}
+else
+{
+	$assocParam = '';
+}
```
Comment (andrepereiradasilva) [inline components/com_content/views/article/tmpl/default.php]: IMO just would be better

``` php
$assocParam = JLanguageAssociations::isEnabled() ? $params->get('show_associations', '') : '';
```

File: language/en-GB/en-GB.ini [hunk 2]
```diff
@@ -52,6 +52,7 @@ JALL="All"
 JALL_LANGUAGE="All"
 JAPPLY="Save"
 JARCHIVED="Archived"
+JASSOCIATIONS="Associations:&nbsp;"
```
Comment (andrepereiradasilva) [inline language/en-GB/en-GB.ini]: should `:&nbsp;` be in a language var ... ?

File: components/com_content/helpers/association.php [hunk 3]
```diff
@@ -60,4 +60,53 @@ public static function getAssociations($id = 0, $view = null)
...
+					{
+						$flag = JHtml::_('image', 'mod_languages/' . $language->image . '.gif',
+								$language->title_native, array('title' => $language->title_native), true);
+						$url  = JRoute::_('<a href="' . JRoute::_($value) . '">' . $flag . '</a>') . '&nbsp;';
+						echo $url;
+					}
+				}
+			}
+		}
+	}
```
Comment (andrepereiradasilva) [inline components/com_content/helpers/association.php:112]: we really need an JAssociationsHelper to generate the assoc links everywhere...

File: language/en-GB/en-GB.ini (hunk 2 above)
Comment (infograf768) [reply language/en-GB/en-GB.ini]: yes or we can add in the new method $url before JRoute
as i figured you would want to do a new general helper, this is for now a POC

File: components/com_content/views/article/tmpl/default.php (hunk 1 above)
Comment (infograf768) [reply components/com_content/views/article/tmpl/default.php]: will do

File: components/com_content/helpers/association.php (hunk 3 above)
Comment (infograf768) [reply components/com_content/helpers/association.php:112]: indeed

**Pull request closed without merging**